### ISSUE TITLE
[bug] compile bug, set_varval

### ISSUE THREAD

**Issue by bymaximus** [slact/nchan#231]
Getting error while compiling latest version
```
/root/nginx-felipe/nchan/src/nchan_variables.c: In function ‘nchan_stub_status_shared_memory_used’:
/root/nginx-felipe/nchan/src/nchan_variables.c:168:17: error: pointer targets in passing argument 2 of ‘set_varval’ differ in signedness [-Werror=pointer-sign]
   set_varval(v, "?", 1);
                 ^
/root/nginx-felipe/nchan/src/nchan_variables.c:29:13: note: expected ‘u_char * {aka unsigned char *}’ but argument is of type ‘char *’
 static void set_varval(ngx_http_variable_value_t *v, u_char *data, size_t len) {
             ^
cc -c -fPIC -pipe  -O -W -Wall -Wpointer-arith -Wno-unused-parameter -Werror -g -g -O2 -fstack-protector --param=ssp-buffer-size=4 -Wformat -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -DNDK_SET_VAR -DNDK_SET_VAR -I src/core -I src/event -I src/event/modules -I src/os/unix -I src/http/modules/perl -I /root/nginx-felipe/njs/nginx/../nxt -I /root/nginx-felipe/njs/nginx/../njs -I /root/nginx-felipe/ngx_devel_kit-0.3.0/objs -I objs/addon/ndk -I /root/nginx-felipe/ngx_devel_kit-0.3.0/src -I /root/nginx-felipe/ngx_devel_kit-0.3.0/objs -I objs/addon/ndk -I /root/nginx-felipe/lua-nginx-module/src/api -I /usr/include/lua5.1 -I /usr/local/include/libdrizzle-1.0 -I /usr/local/include/libdrizzle-1.0 -I /root/nginx-felipe/nchan/src -I /usr/include/libxml2 -I objs -I src/http -I src/http/modules -I src/http/v2 -I /root/nginx-felipe/ngx_devel_kit-0.3.0/src -I src/mail -I src/stream \
        -o objs/addon/src/nchan_module.o \
        /root/nginx-felipe/nchan/src/nchan_module.c
cc1: all warnings being treated as errors
objs/Makefile:3190: recipe for target 'objs/addon/src/nchan_variables.o' failed
make[1]: *** [objs/addon/src/nchan_variables.o] Error 1
make[1]: *** Waiting for unfinished jobs....
make[1]: Leaving directory '/root/nginx-felipe/nginx'
Makefile:8: recipe for target 'build' failed
make: *** [build] Error 2
```

Solved with change
From nchan_variables.c:168
```
set_varval(v, "?", 1);
```
To
```
set_varval(v, (u_char *)'?', 1);
```

**Comment by slact** [comment]
Yeah, I noticed this a few minutes ago. However, I have a better fix in mind...

**Comment by slact** [comment]
Fix in  3ef9fea.

**Comment by bymaximus** [comment]
Awesome, thank you
